crates/ev-revm/src/factory.rs

```diff
@@ -0,0 +1,169 @@
+    type Tx = TxEnv;
+    type Error<DBError: std::error::Error + Send + Sync + 'static> =
+        EVMError<DBError, InvalidTransaction>;
+    type HaltReason = HaltReason;
+    type Spec = SpecId;
+    type Precompiles = PrecompilesMap;
+
+    fn create_evm<DB: Database>(
+        &self,
+        db: DB,
+        evm_env: EvmEnv<Self::Spec>,
+    ) -> Self::Evm<DB, NoOpInspector> {
+        let inner = self.inner.create_evm(db, evm_env);
+        EvEvm::from_inner(inner, self.redirect, false)
+    }
+
+    fn create_evm_with_inspector<DB: Database, I: Inspector<Self::Context<DB>>>(
+        &self,
+        db: DB,
+        input: EvmEnv<Self::Spec>,
+        inspector: I,
+    ) -> Self::Evm<DB, I> {
+        let inner = self.inner.create_evm_with_inspector(db, input, inspector);
+        EvEvm::from_inner(inner, self.redirect, true)
+    }
+}
+
+/// Wraps an [`EthEvmConfig`] so that it produces [`EvEvm`] instances.
+pub fn with_ev_handler<ChainSpec>(
+    config: EthEvmConfig<ChainSpec, EthEvmFactory>,
+    redirect: Option<BaseFeeRedirect>,
+) -> EthEvmConfig<ChainSpec, EvEvmFactory<EthEvmFactory>> {
+    let EthEvmConfig {
+        executor_factory,
+        block_assembler,
+    } = config;
+    let wrapped_factory = EvEvmFactory::new(executor_factory.evm_factory().clone(), redirect);
+    let new_executor_factory = EthBlockExecutorFactory::new(
+        executor_factory.receipt_builder().clone(),
+        executor_factory.spec().clone(),
+        wrapped_factory,
+    );
+
+    EthEvmConfig {
+        executor_factory: new_executor_factory,
+        block_assembler,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use alloy_primitives::{address, Bytes, TxKind, U256};
+    use reth_revm::{
+        revm::{
+            context_interface::result::ExecutionResult,
+            database::{CacheDB, EmptyDB},
+            primitives::{spec::SpecId, AccountInfo, KECCAK_EMPTY},
+            state::EvmState,
+            State,
+        },
+        EvmEnv,
+    };
+
+    #[test]
+    fn factory_applies_base_fee_redirect() {
+        let sink = address!("0x00000000000000000000000000000000000000fe");
+        let beneficiary = address!("0x00000000000000000000000000000000000000be");
+        let caller = address!("0x00000000000000000000000000000000000000ca");
+        let target = address!("0x00000000000000000000000000000000000000aa");
+
+        let mut state = State::builder()
+            .with_database(CacheDB::<EmptyDB>::default())
+            .with_bundle_update()
+            .build();
+
+        state.insert_account(
+            caller,
+            AccountInfo {
+                balance: U256::from(1_000_000_000u64),
+                nonce: 0,
+                code_hash: KECCAK_EMPTY,
+                code: None,
+            },
+        );
+
+        let mut evm_env = EvmEnv::default();
+        evm_env.cfg_env.chain_id = 1;
+        evm_env.cfg_env.spec_id = SpecId::CANCUN;
```

Comment:
I do think this needs to be `spec` to compile?
```suggestion
        evm_env.cfg_env.spec = SpecId::CANCUN;
```